tests/test_multipart.py

```diff
@@ -167,6 +168,67 @@ def test_multipart_manager_generate_presigned_part_url_content_length(
     assert 'content-length' in upload_url
 
 
+def test_multipart_manager_generate_presigned_complete_url(multipart_manager: MultipartManager):
+    upload_url = multipart_manager._generate_presigned_complete_url(
+        UploadCompletion(object_key='new-object', upload_id='fake-upload-id', parts=[])
+    )
+
+    assert isinstance(upload_url, str)
+
+
+def test_multipart_manager_marshal_complete_body(multipart_manager: MultipartManager):
+    body = multipart_manager._marshal_complete_body(
+        UploadCompletion(
+            object_key='new-object',
+            upload_id='fake-upload-id',
+            parts=[
+                PartCompletion(part_number=1, size=1, etag='fake-etag-1'),
+                PartCompletion(part_number=2, size=2, etag='fake-etag-2'),
+            ],
+        )
+    )
+
+    assert body == (
+        '<CompleteMultipartUpload xmlns="http://s3.amazonaws.com/doc/2006-03-01/">'
+        '<Part><PartNumber>1</PartNumber><ETag>fake-etag-1</ETag></Part>'
+        '<Part><PartNumber>2</PartNumber><ETag>fake-etag-2</ETag></Part>'
+        '</CompleteMultipartUpload>'
+    )
+
+
+@pytest.mark.parametrize('file_size', [10, mb(10), mb(12)], ids=['10B', '10MB', '12MB'])
+def test_multipart_manager_get_upload_size(multipart_manager: MultipartManager, file_size: int):
+    # Upload an object
```

Comment:
This entire setup code could be replaced with a `storage.save` to put a fully formed object into S3.